lib/ansible/modules/network/fortios/fortios_address_group.py

```diff
@@ -0,0 +1,236 @@
+    argument_spec.update(fortios_argument_spec)
+
+    #decalre module
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=fortios_required_if,
+    )
+
+    #check params
+    if is_invalid_name(module.params['name']):
+        module.fail_json(msg="Bad name argument value %s, must contain only letters, digits, -, _, ." % (module.params['name']))
+
+    for member in module.params['member']:
+        if is_invalid_name(member):
+            module.fail_json(msg="Bad member argument value %s, must contain only letters, digits, -, _, ." % (member))
+
+    #prepare return dict
+    result = dict(changed=False)
+
+    # fail if PYFG not present
+    if not HAS_PYFG:
+        module.fail_json(msg='Could not import the python library pyFG required by this module')
+
+    #define device
+    f = FortiOS( module.params['host'],
+        username=module.params['username'],
+        password=module.params['password'],
+        timeout=module.params['username'],
+        vdom=module.params['vdom'])
+
+    path = 'firewall addrgrp'
+
+    #connect
+    try:
+        f.open()
+    except Exception:
+        e = get_exception()
+        module.fail_json(msg='Error connecting device. %s' % e)
+
+    #get  config
+    try:
+        f.load_config(path=path)
+        result['firewall_address_config'] = f.running_config.to_text()
+    except Exception:
+        f.close()
+        e = get_exception()
+        module.fail_json(msg='Error reading running config. %s' % e)
+
+
+    #load group member list if group exists
+    group_members = []
+    try:
+        group_members = shlex.split(f.running_config[path][module.params['name']].get_param('member'))
+    except:
+        pass
+
+    #generate target group list
+    if module.params['state'] == 'absent':
+        for member in module.params['member']:
+            if member in group_members:
+                group_members.remove(member)
+    else:
+        #state == present
+        for member in module.params['member']:
+            if member not in group_members:
+                group_members.append(member)
+
+    #delete group if empty
+    if len(group_members) == 0:
+        f.candidate_config[path].del_block(module.params['name'])
+    else:
+        #create addrgrp if not exist (when there is no address group)
+        if path not in f.candidate_config.to_text():
+            new_conf = FortiConfig(path,'config')
+            f.candidate_config[path] = new_conf
+
+        #process changes
+        new_grp = FortiConfig(module.params['name'], 'edit')
+        new_grp.set_param('member', " ".join(group_members) )
+
+        if module.params['comment'] is not None:
+            new_grp.set_param('comment', '"%s"' % (module.params['comment']))
+
+        #add to candidate config
+        f.candidate_config[path][module.params['name']] = new_grp
+
+    #compare config
+    change_string = f.compare_config()
```

Comment:
There is some code duplicated between this pull-request and other FortisOS pull-requests ([`fortios_ipv4_policy`](https://github.com/ansible/ansible/pull/21849#discussion_r103169920) and [`fortios_address`](#21542)):

- block calling `compare_config` method
- block calling `commit` method

Either an helper method or a parent class (see for example [cloudstack](https://github.com/ansible/ansible/tree/devel/lib/ansible/modules/cloud/cloudstack) modules) must be used instead of duplicating the common code.